src/Compiler/test/Microsoft.AspNetCore.Razor.Test.Common/Language/IntegrationTests/RazorIntegrationTestBase.cs

```diff
@@ -398,12 +403,14 @@ protected class CompileToAssemblyResult
 
     private class CompilationFailedException : XunitException
     {
-        public CompilationFailedException(Compilation compilation)
+        public CompilationFailedException(Compilation compilation, ImmutableArray<Diagnostic>? diagnostics = null)
```

Comment:
Is there a reason we chose to use nullable types here vs default for the missing case?

Reply:
No reason, just an oversight during refactoring (https://github.com/dotnet/razor-compiler/commit/3ab81c7ac52424eb7a8e9d3a1e090ce501d98cf2), but you're right, using `default` is better: https://github.com/dotnet/razor-tooling/pull/6814/commits/8e00ce10dd3e3a33dc35ea82e7ea4b2321170e84